doc/Addendum-NX-GZIP-for-PowerVM.txt

```diff
@@ -0,0 +1,317 @@
+  __u16 reserved; __u64 flags;
+  __u64 reserved2[6];
+};
+
+  User-space must set flags = VAS_WIN_QOS_CREDITS in the tx_win_open
+  ioctl call.
+
+  Number of credits available in LPAR will be available in sysfs.
+  Adding the following sysfs interfaces to the user-space:
+
+/sys/kernel/vas/VasCaps/VDefGzip:    (Called default GZIP capabilities)
+	avail_lpar_creds  target_lpar_creds  used_lpar_creds      
+
+  On our development system with 16 cores and dedicated: these entry
+  values are 320 320 0 respectively
+
+/sys/kernel/vas/VasCaps/VQosGzip    (Called QoS GZIP capabilities)
+	avail_lpar_creds   target_lpar_creds  used_lpar_creds
+ 
+
+  VAS instances: There is no concept of VAS ID on PowerVM. Instead the
+  PowerVM must receive a domain ID (based on the CPU ID) in the
+  H_ALLOCATE_VAS_WINDOW hcall to allocate a VAS window on the specific
+  chip and node.  Domain ID to VAS ID mapping cannot be easily
+  determined by the user-space. Therefore, in the Linux driver
+  implementation When user-space sets vas_id= -1, the domain ID is set
+  by the Linux kernel for the cpu core the process is executing on.
+  This interface was added to support NUMA node affinity as found on
+  PowerNV.  In a future release, we are considering passing a specific
+  core ID from the user-space with the existing user-space interface
+  where vas_id may be treated as cpuid.
+
+  __s16 vas_id / CPU_id; /* specific instance of vas or -1 for default */
+
+
+Limits 
+------
+
+  PowerVM restricts the maximum request buffer length to 1MB: PowerVM
+  returns this information in the NX-gzip capabilities hcall, and
+  Linux provides these values to the user-space through sysfs:
+
+/sys/devices/vio/ibm,compression-v1/NxGzCaps
+min_compress_len   --- Recommended minimum compress length in bytes
+min_decompress_len --- Recommended minimum decompress length in bytes
+req_max_processed_len ---  Maximum processed length in bytes
+On our development system, these values are: 
+32768
+32768
+1048576
+
+Max request buffer length is enforced. Minimums are not enforced.
+
+
+Page Faults
+-----------
+
+  Handling of nx-gzip caused page faults is same between PowerVM and
+  PowerN. Refer to the ERR_NX_AT_FAULT condition code in the libnxz
+  source [LIBNXZ]
+
+
+To Do
+-----
+
+  Logical partition migration (LPM) support: all the existing windows
+  have to be closed on the source LPAR and reopen on destination LPAR
+  without the user space knowledge.
+
+  DLPAR support: Whenever core resources are changed, number of
+  available credits will be changed and this information is available
+  with VAS capabilities. In the current implementation, we get these
+  capabilities initially but do not change them. With DLPAR
+  capabilities must be updated.
+
+  Debugfs: provide window status and information through debugfs
+
+
+References
+----------
+
+[P1]
+https://patchwork.ozlabs.org/project/linuxppc-dev/list/?series=238878

```

Comment:
The link is empty, it should be https://patchwork.ozlabs.org/project/linuxppc-dev/list/?series=238878&state=*